applier/applier_test.go

```diff
@@ -500,6 +504,219 @@ func TestApplier_ApplyLayers(t *testing.T) {
+						BlobID: "sha256:932da51564135c98a49a34a193d6cd363d8fa4184d957fde16c9d8527b3f3b02",
+					},
+					Returns: cache.LocalArtifactCacheGetBlobReturns{
+						BlobInfo: types.BlobInfo{
+							SchemaVersion: 1,
+							Digest:        "sha256:932da51564135c98a49a34a193d6cd363d8fa4184d957fde16c9d8527b3f3b02",
+							DiffID:        "sha256:a187dde48cd289ac374ad8539930628314bc581a481cdb41409c9289419ddb72",
+							PackageInfos: []types.PackageInfo{
+								{
+									FilePath: "var/lib/dpkg/status.d/tzdata",
+									Packages: []types.Package{
+										{
+											Name:       "tzdata",
+											Version:    "2019a-0+deb9u1",
+											SrcName:    "tzdata",
+											SrcVersion: "2019a-0+deb9u1",
+										},
+									},
+								},
+							},
+							CustomResources: []types.CustomResource{
+								{
+									Type:     "type-A",
+									FilePath: "var/lib/dpkg/status.d/tzdata",
+									Data: dummyData{
+										data: "Common Package type-A var/lib/dpkg/status.d/tzdata",
+									},
+								},
+								{
+									Type:     "type-B",
+									FilePath: "var/lib/dpkg/status.d/tzdata",
+									Data: dummyData{
+										data: "Common Package type-B var/lib/dpkg/status.d/tzdata",
+									},
+								},
+							},
+						},
+					},
+				},
+				{
+					Args: cache.LocalArtifactCacheGetBlobArgs{
+						BlobID: "sha256:dffd9992ca398466a663c87c92cfea2a2db0ae0cf33fcb99da60eec52addbfc5",
+					},
+					Returns: cache.LocalArtifactCacheGetBlobReturns{
+						BlobInfo: types.BlobInfo{
+							SchemaVersion: 1,
+							Digest:        "sha256:dffd9992ca398466a663c87c92cfea2a2db0ae0cf33fcb99da60eec52addbfc5",
+							DiffID:        "sha256:aad63a9339440e7c3e1fff2b988991b9bfb81280042fa7f39a5e327023056819",
+							Applications: []types.Application{
+								{
+									Type:     "composer",
+									FilePath: "php-app/composer.lock",
+									Libraries: []types.Package{
+										{
+											Name:    "guzzlehttp/guzzle",
+											Version: "6.2.0",
+										},
+										{
+											Name:    "symfony/process",
+											Version: "v4.2.7",
+										},
+									},
+								},
+								{
+									Type:     "composerCustomAnalyzer",
+									FilePath: "php-app/composer.lock",
+									Libraries: []types.Package{
+										{
+											Name:    "guzzlehttp/guzzle",
+											Version: "6.2.0",
+										},
+										{
+											Name:    "symfony/process",
+											Version: "v4.2.7",
+										},
+									},
+								},
+							},
+							CustomResources: []types.CustomResource{
+								{
+									Type:     "type-A",
+									FilePath: "php-app/composer.lock",
+									Data: dummyData{
+										data: "Common Application type-A php-app/composer.lock",
+									},
+								},
+								{
+									Type:     "type-B",
+									FilePath: "php-app/composer.lock",
+									Data: dummyData{
+										data: "Common Application type-B php-app/composer.lock",
+									},
+								},
```

Comment:
We should check if replacing a previous layer works or not. When the type and file path are the same, the upper layer must replace CustomResources in the previous layer. In this case, the previous layer has `"var/lib/dpkg/status.d/tzdata"` with `type-B`. We should have the same pair with different data.